frontend/request.go

```diff
@@ -135,22 +140,72 @@ func marshalDockerfile(ctx context.Context, dt []byte, opts ...llb.ConstraintsOp
 	return st.Marshal(ctx)
 }
 
-func MaybeSign(ctx context.Context, client gwclient.Client, st llb.State, spec *dalec.Spec, targetKey string) (llb.State, error) {
-	if signingDisabled(client) {
-		return st, nil
+func getSigningConfigFromContext(ctx context.Context, client gwclient.Client, cfgPath string, sOpt dalec.SourceOpts) (*dalec.PackageSigner, error) {
+	configCtxName := dockerui.DefaultLocalNameContext
+	if cn := getSignConfigCtxName(client); cn != "" {
+		configCtxName = cn
 	}
 
-	signer := spec.GetSigner(targetKey)
-	if signer == nil {
-		return st, nil
+	sc := dalec.SourceContext{Name: configCtxName}
+	signConfigState, err := sc.AsState([]string{cfgPath}, nil, sOpt)
+	if err != nil {
+		return nil, err
 	}
 
-	signed, err := forwardToSigner(ctx, client, signer, st)
+	scDef, err := signConfigState.Marshal(ctx)
 	if err != nil {
-		return llb.Scratch(), err
+		return nil, err
+	}
+
+	res, err := client.Solve(ctx, gwclient.SolveRequest{
+		Definition: scDef.ToPB(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ref, err := res.SingleRef()
+	if err != nil {
+		return nil, err
+	}
+
+	dt, err := ref.ReadFile(ctx, gwclient.ReadRequest{
+		Filename: cfgPath,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var pc dalec.PackageConfig
+	if err := yaml.Unmarshal(dt, &pc); err != nil {
+		return nil, err
 	}
 
-	return signed, nil
+	return pc.Signer, nil
+}
+
+func MaybeSign(ctx context.Context, client gwclient.Client, st llb.State, spec *dalec.Spec, targetKey string, sOpt dalec.SourceOpts) (llb.State, error) {
+	if signingDisabled(client) {
+		return st, nil
+	}
+
+	switch cfgPath := getUserSignConfigPath(client); cfgPath {
```

Comment:
After #322 is merged, we should add a warning when the build-args are overwriting the signing in the spec.

Reply:
sgtm, will wait till that one's merged

Reply:
That one is in now. Let's also make sure to bring back the tests that check to make sure the warning is present 😅 🚀

Reply:
Merged that, updated this

Reply:
done

Reply:
Looks like tests are missing the checks for the warning messages (see: `WithSolveStatusFn`)

Reply:
I need to slow down!

Reply:
done